runtime/parachains/src/system_token_manager.rs

```diff
@@ -186,939 +123,936 @@ pub mod pallet {
+		//
+		// Origin:
+		// ** Root(Authorized) privileged call **
+		//
+		// Params:
+		// - wrapped: `Wrapped` system token id expected to be deregistered
+		//
+		// Logic:
+		// DMP
+		// - Set`sufficient` to `false`
+		// - 'unlink' the system token with local asset
 		pub fn deregister_wrapped_system_token(
 			origin: OriginFor<T>,
-			wrapped_system_token_id: SystemTokenId,
+			wrapped: SystemTokenId,
 		) -> DispatchResult {
-			ensure_root(origin.clone())?;
-			ensure!(
-				SystemTokenOnParachain::<T>::contains_key(&wrapped_system_token_id),
-				Error::<T>::WrappedSystemTokenNotRegistered
-			);
-
-			let system_token_id = SystemTokenOnParachain::<T>::get(&wrapped_system_token_id)
-				.ok_or("original system token id should be obtained")?;
-
-			SystemTokenOnParachain::<T>::remove(&wrapped_system_token_id);
-			if SystemTokenOnParachainByParaId::<T>::contains_key(
-				wrapped_system_token_id.clone().para_id,
-			) {
-				let _ = SystemTokenOnParachainByParaId::<T>::try_mutate_exists(
-					wrapped_system_token_id.clone().para_id,
-					|maybe_system_tokens| -> Result<(), DispatchError> {
-						let system_tokens =
-							maybe_system_tokens.as_mut().ok_or(Error::<T>::Unknown)?;
-						system_tokens.retain(|x| *x != wrapped_system_token_id.clone());
-						Ok(())
-					},
-				);
-			}
-
-			if ParaIdsBySystemToken::<T>::contains_key(system_token_id.clone()) {
-				let _ = ParaIdsBySystemToken::<T>::try_mutate_exists(
-					system_token_id.clone(),
-					|maybe_para_ids| -> Result<(), DispatchError> {
-						let para_ids = maybe_para_ids.as_mut().ok_or(Error::<T>::Unknown)?;
-						para_ids.retain(|x| *x != wrapped_system_token_id.clone().para_id.into());
-						Ok(())
-					},
-				);
-			}
-
-			WrappedSystemTokenProperties::<T>::remove(&wrapped_system_token_id);
-
-			// Deregister wrapped system token in relay chain
-			if wrapped_system_token_id.clone().para_id == 0.into() {
-				let _ = pallet_assets::pallet::Pallet::<T>::set_sufficient_with_unlink_system_token(
-					origin.clone(),
-					wrapped_system_token_id.clone().asset_id.into(),
-					false,
-				);
-			} else {
-				// DMP calls
-				Self::dmp_set_sufficient_with_unlink(wrapped_system_token_id.clone(), false);
-			}
+			ensure_root(origin)?;
+			Self::try_deregister(SystemTokenType::Wrapped(wrapped, false))?;
+			Self::try_set_sufficient_and_unlink(wrapped.clone(), false)?;
 
-			Self::deposit_event(Event::<T>::WrappedSystemTokenDeregistered {
-				wrapped_system_token_id,
-				system_token_id,
-			});
+			Self::deposit_event(Event::<T>::WrappedSystemTokenDeregistered { wrapped });
 
 			Ok(())
 		}
 
 		#[pallet::call_index(4)]
 		#[pallet::weight(1_000)]
-		/// Deregister the system token.
+		// Description:
+		// Update the weight of system token, which can affect on PoT vote
+		//
+		// Origin:
+		// ** Root(Authorized) privileged call **
+		//
+		// Params:
+		// - system_token_id: System token id expected to be updated
```

Comment:
`// - system_token_id: System token id expected to be updated` -> 
`// - original: System token id expected to be updated`

typo